app/uk/gov/hmrc/gform/validation/ValidationService.scala

```diff
@@ -439,31 +439,32 @@ class ComponentsValidator(
     maxWhole: Int,
     maxFractional: Int,
     mustBePositive: Boolean): ValidatedType = {
-    val WholeShape = "([+-]?)(\\d+)[.]?".r
-    val FractionalShape = "([+-]?)(\\d*)[.](\\d+)".r
+    val WholeShape = "([+-]?)(\\d+(,\\d{3})*?)[.]?".r
+    val FractionalShape = "([+-]?)(\\d*(,\\d{3})*?)[.](\\d+)".r
     (TextConstraint.filterNumberValue(value), maxFractional, mustBePositive) match {
-      case (WholeShape(_, whole), _, _) if whole.size > maxWhole =>
+      case (WholeShape(_, whole, _), _, _) if whole.size > maxWhole =>
```

Comment:
This counts commas to number size. It seems wrong.

We accept `1234567890` but reject `1,234,567,890`. This is not consistent.